src/praisonai/tests/unit/test_langextract_sink.py

```
@@ -342,5 +342,102 @@ def test_observe_invalid_provider_error(self):
+            data={},
+        ))
+
+        types = [e.event_type for e in sink._events]
+        assert "agent_start" in types
+        assert "tool_start" in types
+        assert "tool_end" in types
+        assert "output" in types
+        assert "agent_end" in types
+        assert sink._source_text == "Write a haiku"
+
+    @pytest.mark.skipif(not _langextract_available(), reason="langextract not installed")
```

Ruff (0.15.10)

[error] Undefined name `_langextract_available`

(F821)